lib/core/versions/latest/OperationProcessor.ts

```diff
@@ -113,8 +114,12 @@ export default class OperationProcessor implements IOperationProcessor {
       lastOperationTransactionNumber: anchoredOperationModel.transactionNumber
     };
 
-    // Ensure actual delta hash matches expected delta hash.
-    const isMatchingDelta = Multihash.isValidHash(operation.encodedDelta, operation.suffixData.deltaHash);
+    // Verify the delta hash against the expected delta hash.
+    const expectedDeltaHash = operation.delta ? Encoder.encode(JsonCanonicalizer.canonicalizeAsBuffer({
```

Comment:
Suggesting two changes:
1. Output is not a hash, rename.
2. Should not encode just to be able to use `Multihash.isValidHash`, make `Multihash.verify()` public and use that. Though the name should be better now that it's public. Or maybe even better introduce a new method `Multihash.canonicalizeAndVerifyHash()`.
```suggestion
    const deltaPayload = operation.delta ? JsonCanonicalizer.canonicalizeAsBuffer({
```